File: client/lib/data/entities/knockout_position.dart
import 'package:solidtrade/data/entities/base/base_entity.dart';
import 'package:solidtrade/data/entities/base/base_position.dart';

class KnockoutPosition implements IBaseEntity, IPosition {
  @override
  final int id;
  @override
  final DateTime createdAt;
  @override
  final DateTime updatedAt;

  final String isin;
  final double buyInPrice;
  final double numberOfShares;

  const KnockoutPosition({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    required this.isin,
    required this.buyInPrice,
    required this.numberOfShares,
  });

  factory KnockoutPosition.fromJson(Map<String, dynamic> json) {
    return KnockoutPosition(
      id: json["id"],
      createdAt: DateTime.parse(json["createdAt"]),
      updatedAt: DateTime.parse(json["updatedAt"]),
      isin: json["isin"],
      buyInPrice: json["buyInPrice"],
      numberOfShares: json["numberOfShares"],
    );
  }
}

File: server/Domain/Entities/KnockoutPosition.cs
using System.ComponentModel.DataAnnotations;
using Domain.Common;
using Domain.Common.Position;
using Domain.Entities.Base;

namespace Domain.Entities;

public class KnockoutPosition : BaseEntity, IPosition
{
    [Required]
    public Portfolio Portfolio { get; set; }

    [Required]
    [MaxLength(12)]
    public string Isin { get; set; }
        
    [Required]
    [Range(DomainConstants.MinimumNumberOfShares, int.MaxValue)]
    public decimal NumberOfShares { get; set; }
        
    [Required]
    public decimal BuyInPrice { get; set; }
}